File: src/main/java/com/kelsos/mbrc/Data/MusicTrack.java
package com.kelsos.mbrc.data;

public class MusicTrack {
    private String title;
    private String artist;

    public MusicTrack(String artist, String title) {
        super();
        this.title = title;
        this.artist = artist;
	}

    public String getArtist() {
        return artist;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

	@Override
	public boolean equals(Object o)
	{
		if(o instanceof MusicTrack)
		{
			MusicTrack track = (MusicTrack)o;
			if(track.getTitle().equals(this.title)&&track.getArtist().equals(this.artist))
				return true;
		}
		return false;
	}
}

File: app/src/main/kotlin/com/kelsos/mbrc/data/MusicTrack.kt
package com.kelsos.mbrc.data

import com.fasterxml.jackson.databind.JsonNode

class MusicTrack {
  var title: String? = null
  var artist: String? = null
    private set
  var position: Int = 0
    private set

  constructor(node: JsonNode) {
    this.artist = node.path("Artist").textValue()
    this.title = node.path("Title").textValue()
    this.position = node.path("Position").intValue()
  }

  constructor(artist: String, title: String) {
    this.artist = artist
    this.title = title
    position = 0
  }

  override fun equals(o: Any?): Boolean {
    var rValue = false
    if (o is MusicTrack) {
      if (o.title == this.title && o.artist == this.artist) {
        rValue = true
      }
    }
    return rValue
  }

  override fun hashCode(): Int {
    return title!!.hashCode() + artist!!.hashCode()
  }
}
